packages/server/src/profile/profile.service.ts

```diff
@@ -23,27 +26,47 @@ export class ProfileService {
         profId: userId,
       },
     });
+    const lastRegisteredSemester = lastRegistered?.lastRegisteredSemester;
 
     if (!profCourses) return; // not a professor
 
-    const profCourseSem =
-      profCourses.sectionGroups[0] && profCourses.sectionGroups[0].semester;
-
     const pendingCourses = [];
 
-    if (lastRegistered?.lastRegisteredSemester !== profCourseSem) {
+    for (const c of profCourses.sectionGroups) {
       // current semester does not match last time prof registered, there may be pending courses
-      for (const c of profCourses.sectionGroups) {
-        if (c.crns.length !== 0) {
-          const courseCRN = c.crns[0];
-          const profCourse = await this.loginCourseService.courseCRNToCourse(
-            courseCRN,
-            c.semester,
-          );
-          if (!profCourse) pendingCourses.push(c);
-        }
+      if (
+        c.crns.length !== 0 &&
+        (!lastRegisteredSemester || // lastRegistered doesnt exist if prof has never registered before
+          !this.isSameRegistrationTime(c.semester, lastRegisteredSemester))
+      ) {
+        const courseCRN = c.crns[0];
+        const profCourse = await this.loginCourseService.courseCRNToCourse(
+          courseCRN,
+          c.semester,
+        );
+        if (!profCourse) pendingCourses.push(c);
       }
     }
+
     return pendingCourses;
   }
+
+  // hacky way to check if two semester codes from Khoury have the same registration window
+  private isSameRegistrationTime(semester1: string, semester2: string) {
+    const year1 = Number(semester1.slice(0, 4));
+    const year2 = Number(semester2.slice(0, 4));
+    const semesterCode1 = semester1.slice(-2);
+    const semesterCode2 = semester2.slice(-2);
+    // we want to treat Summer 1 and Summer Full as the same window
+    // for profs to register classes for both at the same time
+    if (
```

Comment:
Nit: just save the Summer1 and Summer full codes as constants so you don't have to access them twice

Reply:
yes